src/PID.c

```diff
@@ -0,0 +1,22 @@
+#include <main.h>
+
+/**
+ Default PID struct constructor, used for default parameter value
+ These specific values represent a smooth curve with no overshoot
+ */
+const struct PID DEFAULT_PID = { 5, 3, 3 };
+
+/**
+ Gets the output of the input passed through the PID controller for the delta time interval
+ */
+int pid(int input, int *errorSum, const unsigned long * const deltaTime, int target, PID pid) {
```

Comment:
Not sure I understand why on your first point, doesnt that make this less memory intensive and to the second point, will do